theseus/optimizer/linear/tests/test_lu_cuda_sparse_solver.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import pytest  # noqa: F401
+import torch
+
+import theseus.core as theseus
+import theseus.optimizer as thoptim
+import theseus.optimizer.linear as thlin
+
+
+def _build_sparse_mat(batch_size):
+    all_cols = list(range(10))
+    col_ind = []
+    row_ptr = [0]
+    for i in range(12):
+        start = max(0, i - 2)
+        end = min(i + 1, 10)
+        col_ind += all_cols[start:end]
+        row_ptr.append(len(col_ind))
+    data = torch.randn((batch_size, len(col_ind)), dtype=torch.double)
+    return 12, 10, data, col_ind, row_ptr
+
+
+def test_sparse_solver():
```

Comment:
Are we testing unrolling operations? Even w/o the implicit differentiation checks, we could add some basic test for this werein we do a series of `solve()` operations and check some tensor's `.grad` content numerically?

Reply:
+1

Reply:
that can certainly be done!